iotop/ui.py

```diff
@@ -80,29 +82,49 @@ def format_size(options, bytes):
+    if options.prometheus:
+        return format_size(options, size and float(size))
+    else:
+        return format_size(options, size and float(size) / duration) + '/s'
 
 
 def format_stats(options, process, duration):
     # Keep in sync with TaskStatsNetlink.members_offsets and
     # IOTopUI.get_data(self)
-    def delay2percent(delay):  # delay in ns, duration in s
+    def delay2percent(delay, duration):  # delay in ns, duration in s
         return '%.2f %%' % min(99.99, delay / (duration * 10000000.0))
-    if options.accumulated:
-        stats = process.stats_accum
-        display_format = lambda size, duration: format_size(options, size)
-        duration = time.monotonic() - process.stats_accum_timestamp
-    else:
-        stats = process.stats_delta
-        display_format = lambda size, duration: format_bandwidth(
-                                                       options, size, duration)
-    io_delay = delay2percent(stats.blkio_delay_total)
-    swapin_delay = delay2percent(stats.swapin_delay_total)
-    read_bytes = display_format(stats.read_bytes, duration)
-    written_bytes = stats.write_bytes - stats.cancelled_write_bytes
-    written_bytes = max(0, written_bytes)
-    write_bytes = display_format(written_bytes, duration)
-    return io_delay, swapin_delay, read_bytes, write_bytes
+
+    def get_accum_stats():
+        accum_stats = process.stats_accum
+        accum_display_format = lambda size, duration: format_size(options, size)
+        accum_duration = time.monotonic() - process.stats_accum_timestamp
+
+        stats_accum = defaultdict()
+        stats_accum['io_delay'] = delay2percent(accum_stats.blkio_delay_total, accum_duration)
+        stats_accum['swapin_delay'] = delay2percent(accum_stats.swapin_delay_total, accum_duration)
+        stats_accum['read_bytes'] = accum_display_format(accum_stats.read_bytes, accum_duration)
+        written_bytes = accum_stats.write_bytes - accum_stats.cancelled_write_bytes
+        written_bytes = max(0, written_bytes)
+        stats_accum['write_bytes'] = accum_display_format(written_bytes, accum_duration)
+
+        return stats_accum
+
+    def get_delta_stats(delta_duration):
+        delta_stats = process.stats_delta
+        delta_display_format = lambda size, duration: format_bandwidth(options, size, duration)
+
+        stats_delta = defaultdict()
+        stats_delta['io_delay'] = delay2percent(delta_stats.blkio_delay_total, delta_duration)
+        stats_delta['swapin_delay'] = delay2percent(delta_stats.swapin_delay_total, delta_duration)
+        stats_delta['read_bytes'] = delta_display_format(delta_stats.read_bytes, delta_duration)
+        written_bytes = delta_stats.write_bytes - delta_stats.cancelled_write_bytes
+        written_bytes = max(0, written_bytes)
+        stats_delta['write_bytes'] = delta_display_format(written_bytes, delta_duration)
+
+        return stats_delta
+
+
+    return get_accum_stats(), get_delta_stats(duration)
```

Comment:
probably not necessary anymore to return both (accumulated and delta) stats anymore, because the "prometheus" option now only relies on the delta metrics